packages/react-server/src/metrics/metrics.ts

```diff
@@ -0,0 +1,27 @@
+import {Context} from 'koa';
+import compose from 'koa-compose';
+
+import {KoaNextFunction} from '../types';
+
+const START_TIME_STATE_KEY = Symbol('startTime');
+
+async function startStatsd(ctx: Context, next: KoaNextFunction) {
+  ctx.state[START_TIME_STATE_KEY] = process.hrtime();
+  await next();
+}
+
+async function middleware(ctx: Context, next: KoaNextFunction) {
+  try {
+    await next();
+  } finally {
+    const [seconds, nanoseconds] = process.hrtime(
+      ctx.state[START_TIME_STATE_KEY],
+    );
+    const ms = seconds * 1000 + nanoseconds / 1e6;
+    const requestTime = Math.round(ms);
+
+    ctx.set('X-React-Server-Request-Time', requestTime.toString());
```

Comment:
nit: could we have this as part of an enum or something rather than an inline string? (Then we can also import it in the test and export it from the package)

Reply:
I'd prefer using [`Server-Timing`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Server-Timing) as a source of truth for all Node=>Rails metrics because it:
- Avoids proliferation of timing header tags every time we think of a new metric
- Allows quilt_rails to find timing data without fishing through arbitrary headers
- Maybe helps us consider what a more generic metrics API looks like (how do we extend it for react-server?  How do consuming apps add their own metrics?)
- Embraces community standards, helping us interface with other libs / tools

Reply:
RE a generic metrics API.  We should be thinking of how to surface info like tracking [cancelled render loops](https://github.com/Shopify/quilt/blob/32f04a86d8ae0f585e96dda5160ffb152073587e/packages/react-effect/src/server.tsx#L41).

Reply:
RE a generic metrics API: it's weird that this does [a chunk of work that's identical to `requestLogger`](https://github.com/Shopify/quilt/blob/master/packages/react-server/src/logger/logger.ts#L37).  What if *another* middleware handled request timing, and Logger / Metrics middleware fished it out of the context?